psydac/linalg/basic.py

```diff
@@ -221,14 +223,354 @@ def codomain(self):
     def dtype(self):
         pass
 
-    @abstractmethod
+    # Function that returns the local matrix corresponding to the linear operator. Returns a scipy.sparse.csr.csr_matrix.
     def tosparse(self):
-        pass
+        """
+        Transforms the linear operator into a matrix, which is stored in sparse csr format.
 
-    @abstractmethod
-    def toarray(self):
-        pass
+        Returns
+        -------
+        out : Numpy.ndarray or scipy.sparse.csr.csr_matrix
+            The matrix form of the linear operator. If ran in parallel each rank gets the local
+            matrix representation of the linear operator.
+        """
+        # v will be the unit vector with which we compute Av = ith column of A.
+        v = self.domain.zeros()
+        # We define a temporal vector
+        tmp2 = self.codomain.zeros()
+        
+        #We need to determine if we are a blockvector or a stencilvector but we are not able to use 
+        #the BlockVectorSpace and StencilVectorSpace classes in here. So we use the following char.
+        #This char is either a b for Blockvectors or an s for Stencilvectors
+        BoS= str(self.domain)[15]
```

Comment:
If in the future this string changes then `tosparse()` would stop working. A potential  alternative would be:
`BoS = 'b' if  hasattr(self.domain, 'spaces') else 's'`